### PR TITLE
[IMP] account: not allow to have different company in move line and a…

### PR DESCRIPTION
**Pull request by var-odoo**
Description of the issue/feature this PR addresses:

- Related task: https://www.odoo.com/web?#id=1970459&model=project.task&view_type=form&menu_id=4720

- Pad: https://pad.odoo.com/p/r.c539df3e17137ae5437f37ccb44f6b38 

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -310,6 +310,10 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
     def create(self, vals):
         move = super(AccountMove, self.with_context(mail_create_nolog=True, check_move_validity=False, partner_id=vals.get('partner_id'))).create(vals)
         move.assert_balanced()
+        for move in self:
+            if move.line_ids:
+                if not all([x.company_id.id == move.company_id.id for x in move.line_ids]):
```
Comment (fmdl) [inline addons/account/models/account_move.py]: use `if any` (less computation)

I think it is better to create an @api.contraints(), no ?

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by fmdl** [comment]
@qdp-odoo, `company_id` of `account.move.line` could be a related field of move, no ?
`company_id = fields.Many2one(related='move_id.company_id', store=True, readonly=True)`

**Comment by qdp-odoo** [comment]
could be but that doesn't change that you want to make sure the company of the journal and of the accounts are the same.

**Comment by qdp-odoo** [comment]
@robodoo retry robot-dude

**Comment by qdp-odoo** [comment]
@var-odoo it looks like the merge cannot succeed. Can you rebase your branch please?

**Comment by var-odoo** [comment]
@qdp-odoo I have rebased the branch. Thanks :+1:

**Comment by qdp-odoo** [comment]
and now, not so surprisingly, the branch is red on the runbot... you need to fix it prior to the merge

**Comment by var-odoo** [comment]
@qdp-odoo The branch is green on the runbot, now you can check. Thanks :+1:

**Comment by qdp-odoo** [comment]
@robodoo r+

**Comment by qdp-odoo** [comment]
@robodoo retry

**Comment by qdp-odoo** [comment]
@var-odoo still not: the merge build raised an error due to this new constraint. Please rebase on master again

**Comment by var-odoo** [comment]
@qdp-odoo I have rebased the branch. Thanks :+1:

**Comment by qdp-odoo** [comment]
the diff didn't change, so I doubt there will be any progress...

@robodoo r+

**Comment by qdp-odoo** [comment]
what a surprise...

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -305,6 +305,13 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
             res['fields']['line_ids']['views']['tree']['fields']['tax_line_id']['domain'] = [('tag_ids', 'in', [self.env.ref(self._context.get('vat_domain')).id])]
         return res
 
+    @api.constrains('company_id')
+    def _check_company_id(self):
+        for move in self:
+            if move.line_ids:
+                if any([x.company_id.id != move.company_id.id for x in move.line_ids]):
+                    raise UserError(_("Cannot create moves for different companies."))
```
Comment (mreficent) [inline addons/account/models/account_move.py]: ValidationError

**Comment by qdp-odoo** [comment]
superseeded by https://github.com/odoo/odoo/pull/34244

**Pull request closed without merging**